src/opentimelineio/typeRegistry.h

```diff
@@ -9,14 +9,25 @@
 #include <algorithm>
 #include <functional>
 #include <map>
+#include <map>
 #include <mutex>
 #include <string>
+#include <unordered_map>
 
 namespace opentimelineio { namespace OPENTIMELINEIO_VERSION {
 
 class SerializableObject;
+class Encoder;
 class AnyDictionary;
 
+// typedefs for the schema downgrading system
+// @TODO: should we make version an int64_t?  That would match what we can
+//        serialize natively, since we only serialize 64 bit signed ints.
+using schema_version_map = std::unordered_map<std::string, int64_t>;
+using label_to_schema_version_map = std::unordered_map<std::string, schema_version_map>;
+
+extern const label_to_schema_version_map CORE_VERSION_MAP;
```

Comment:
What is the "extern" here for?

Reply:
CORE_VERSION_MAP is a global const data structure that is defined in CORE_VERSION_MAP.cpp - the `extern` allows anything importing typeRegistry to access it.

Reply:
Ah, I guess since the data is const it's OK to be global. On Windows there's an extra wrinkle with static data and dynamic libraries, where the app and library each get their own copy of the data. But again since it's const that seems OK.

Reply:
Good to know!

Reply:
If we ever do a __declspec(dllexport) pass, this'll require special treatment. using a function accessor instead of extern const dodges the bullet. example of the complexity it introduces: https://github.com/AcademySoftwareFoundation/Imath/blob/main/src/Imath/ImathExport.h